src/main/scala/edu/berkeley/ce/rockslicing/Block.scala

```diff
@@ -28,12 +47,50 @@ case class Face(normalVec: (Double,Double,Double), distance: Double, phi: Double
     val newD = if (math.abs(d) > NumericUtils.EPSILON) d else 0.0
     val newPhi = if (math.abs(phi) > NumericUtils.EPSILON) phi else 0.0
     val newCohesion = if (math.abs(cohesion) > NumericUtils.EPSILON) cohesion else 0.0
-    Face((newA,newB,newC), newD, newPhi, newCohesion)
+    Face((newA,newB,newC), newD, newPhi, newCohesion, processorJoint)
   }
 }
 
 object Block {
   /**
+    * Compare two blocks for equality within specified tolerance
+    * @param block1 First input block
+    * @param block2 Second input block
+    * @return True if blocks are equal, otherwise false
+    */
+  def compareBlocks(block1: Block, block2: Block, tolerance: Double=NumericUtils.EPSILON):
```

Comment:
Actually, what I was suggesting was that you move this from the companion object into the class itself. So instead of `Block.compareBlocks(block1, block2)` the code would look more like `block1.compareTo(block2)`.

Up to you if we go ahead and do that, but yeah, at least a name change to something like `approximateEquals` is a good idea.